common/src/main/scala/hmda/util/QuarterTimeBarrier.scala

```diff
@@ -0,0 +1,88 @@
+package hmda.util
+
+import java.time.{Clock, LocalDate}
+import hmda.util.BankFilterUtils.config
+import hmda.utils.YearUtils.Period
+
+import java.time.format.DateTimeFormatter
+
+object QuarterTimeBarrier {
+  private  val rulesConfig = Filer.parse(config).fold(error => throw new RuntimeException(s"Failed to parse filing rules in HOCON: $error"), identity)
+
+  // Builtin time format of BASIC_ISO_DATE yyyyMMdd i.e. '20111203'.
+  private val formatter: DateTimeFormatter =DateTimeFormatter.BASIC_ISO_DATE
+  private val  q12020EndDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q1.endDayOfYear)
+  private val  q22020EndDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q2.endDayOfYear)
+  private val  q32020EndDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q3.endDayOfYear)
+
+  private val  q12021EndDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q1.endDayOfYear)
+  private val  q22021EndDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q2.endDayOfYear)
+  private val  q32021EndDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q3.endDayOfYear)
+
+  private val  q12022EndDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q1.endDayOfYear)
+  private val  q22022EndDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q2.endDayOfYear)
+  private val  q32022EndDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q3.endDayOfYear)
+
+  private val  q12020StartDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q1.startDayOfYear)
+  private val  q22020StartDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q2.startDayOfYear)
+  private val  q32020StartDate = LocalDate.ofYearDay(2020,rulesConfig.qf.q3.startDayOfYear)
+
+  private val  q12021StartDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q1.startDayOfYear)
+  private val  q22021StartDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q2.startDayOfYear)
+  private val  q32021StartDate = LocalDate.ofYearDay(2021,rulesConfig.qf.q3.startDayOfYear)
+
+  private val  q12022StartDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q1.startDayOfYear)
+  private val  q22022StartDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q2.startDayOfYear)
+  private val  q32022StartDate = LocalDate.ofYearDay(2022,rulesConfig.qf.q3.startDayOfYear)
+
+  
+  def actionTakenInQuarterRange(actionTakenDate: Int, period: Period):Boolean= {
```

Comment:
might be worth it to add implicit class like:
```
  implicit class ImprovedLocalDate(date: LocalDate) {
    def isOnOrBefore(compareDate: LocalDate): Boolean = {
      date.isBefore(compareDate) || date.isEqual(compareDate)
    }
    def isBetween(start: LocalDate, end: LocalDate): Boolean = {
      date.isAfter(start) && date.isOnOrBefore(end)
    }
  }
```
then the match logics can be like `actionTakenDateLocal.isOnOrBefore(q12020EndDate)`, or `actionTakenDateLocal.isBetween(q12020EndDate, q22020EndDate)`
this way code might be descriptive enough that there wouldn't be a need for comments